### PR TITLE
use uv in setup-env to speed up virtualenv creation

### PR DESCRIPTION
**Pull request by ilaner**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -116,6 +116,7 @@ klara = { version = "^0.6.3", optional = true }
 
 pre-commit = "^3.5.0"
 lxml = "^5.0.0"
+uv = "^0.1.5"
```
Comment (dorschw) [inline pyproject.toml:119]:
```suggestion
uv = "^0.1.6"
```
🤡

File: demisto_sdk/commands/setup_env/setup_environment.py [hunk 2]
```diff
@@ -491,23 +492,56 @@ def install_virtualenv(
...
-                check=True,
-            )
-            logger.info(f"Installed {req}")
-        except subprocess.CalledProcessError:
-            logger.warning(f"Could not install {req}, skipping...")
+    env = os.environ.copy()
+    env.update({"VIRTUAL_ENV": str(venv_path)})
+    requirements = [req for req in requirements if req]
+    try:
+        subprocess.run(
```
Comment (dorschw) [inline demisto_sdk/commands/setup_env/setup_environment.py:499]: for readability, let's split to methods: `install_all_using_uv`, `install_single_using_uv`, `install_all_using_pip`

File: demisto_sdk/commands/setup_env/setup_environment.py [hunk 3]
```diff
@@ -460,6 +461,76 @@ def configure_vscode(
...
+            + requirements,
+            check=True,
+            env=env,
+        )
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+def install_one_by_one(requirements: List[str], env: dict, venv_path: Path) -> None:
```
Comment (dorschw) [inline demisto_sdk/commands/setup_env/setup_environment.py:491]: would it be clearer if we changed it to `install_single` and loop outside?

**Review by dorschw — COMMENTED** [review]
Cool!

File: demisto_sdk/commands/setup_env/setup_environment.py [hunk 4]
```diff
@@ -491,23 +562,15 @@ def install_virtualenv(
...
-                    "install",
-                    req,
-                ],
-                check=True,
-            )
-            logger.info(f"Installed {req}")
-        except subprocess.CalledProcessError:
-            logger.warning(f"Could not install {req}, skipping...")
+    env = os.environ.copy()
+    env.update({"VIRTUAL_ENV": str(venv_path)})
```
[the remark this answers is not in the document]
Comment (ilaner) [reply demisto_sdk/commands/setup_env/setup_environment.py:566]: we can have the same env for both `uv` and `pip`. I don't see a reason to seperate them

File: pyproject.toml (hunk 1 above)
Comment (dorschw) [reply pyproject.toml:121]:
```suggestion
uv = "^0.1.24"
```

**Review by dorschw — APPROVED** [review]

**Pull request merged**